warp10/src/main/java/io/warp10/script/functions/SHRINK.java

```diff
@@ -49,13 +49,15 @@ public Object applyOnElement(Object element) {
         if (element instanceof GeoTimeSerie) {
           GeoTimeSerie gts = (GeoTimeSerie) element;
 
-          if (GTSHelper.nvalues(gts) > Math.abs(shrinkto)) {
-            if (shrinkto < 0) {
-              GTSHelper.sort(gts, true);
-            } else {
-              GTSHelper.sort(gts, false);
+          if (shrinkto != Long.MIN_VALUE && shrinkto != Long.MAX_VALUE) {
+            if (GTSHelper.nvalues(gts) > Math.abs(shrinkto)) {
```

Comment:
this check should come first since since it is 99.999...% of the usages
merge this check with the former line with &&